### DISCUSSION TITLE
Contrast accessibility: Is it possible to customize CLI colors?

### DISCUSSION THREAD

**Discussion by slifty** [eslint/eslint#17695] [Q&A] [answered]
I'm having real trouble reading some of the default output of eslint -- specifically the line numbers / rule name:

![image](https://github.com/eslint/eslint/assets/208884/8a3717c4-2d44-4f07-98b0-67b4b12c3a1d)

I've tried setting maximum contrast in my terminal, but even then the line numbers manage to be too dark.  I do value the colors in general (e.g. error) and so I'd love to avoid having to set the `--no-color` flag.  Does eslint provide any way to customize the colors used in the output, or would it be possible to just try to avoid this kind of severe low-contrast issue in the default outputs?

**Comment by slifty** [comment]
I see the code that is adding the colors is here: https://github.com/eslint/eslint/blob/c0b11ddb9f8aacc64c3933b9f278939aa7bea481/lib/cli-engine/formatters/stylish.js#L38-L79

It looks like the issue is `dim` (and to answer my title question, it's not something that can be configured currently; I'll maybe open a feature request just in case it would be considered)

**Reply by nzakas** [reply, marked as answer]
Formatters are completely self-contained functions. ESLint has no idea what is being output, whether there are colors or not, so there isn't any way to implement an option that will "just work" for all ESLint output.

If you don't like the colors in the default formatter, you can always [try a different one](https://www.npmjs.com/search?q=eslint-formatter), or you can [write your own](https://eslint.org/docs/latest/extend/custom-formatters).

**Comment by arkapratimc** [comment]
The functions which log the results in terminal are called formatters. ESLint comes [built-in](https://eslint.org/docs/latest/use/formatters/) with a bunch of those, plus, you can make your own function to log the results - https://eslint.org/docs/latest/extend/custom-formatters
